dotman-plugin/src/main/java/net/minevn/dotman/UpdateChecker.kt

```diff
@@ -0,0 +1,50 @@
+package net.minevn.dotman
+
+import net.minevn.dotman.utils.Utils.Companion.info
+import net.minevn.libs.bukkit.parseJson
+import org.bukkit.entity.Player
+import net.minevn.libs.get
+
+class UpdateChecker {
+    private val plugin = DotMan.instance
+    private val url = "https://api.github.com/repos/MineVN/DotMan/releases/latest"
+    private var releaseVersion : String = ""
+    private val currentVersion = plugin.description.version.trim()
+    private var latestVersion = currentVersion
+    private var latest = false
+
+    fun init() {
+        latest = checkUpdate()
+        val language = plugin.language
+        if (latest) {
+            info(plugin.language.updateLatest)
+        } else {
+            language.updateAvailable
+                .replace("%NEW_VERSION%", "")
+                .replace("%CURRENT_VERSION%", currentVersion)
+                .let { info(it) }
+            info(plugin.language.updateAvailableLink.replace("%URL%", releaseVersion))
+        }
+    }
+
+    // Check for updates when a player logs in
+    fun loginCheckForUpdates(player: Player) {
+        if (plugin.config.checkUpdate && !latest && player.hasPermission("dotman.update")) {
+            val language = plugin.language
+            language.updateAvailable
+                .replace("%NEW_VERSION%", latestVersion)
+                .replace("%CURRENT_VERSION%", currentVersion)
+                .let { player.sendMessage(it) }
```

Comment:
Đây là message chỉ xuất hiện trong giao diện của Admin, nên không cần thiết phải cho vào config.

Reply:
> Đây là message chỉ xuất hiện trong giao diện của Admin, nên không cần thiết phải cho vào config.

E le altre lingue?
https://github.com/EssentialsX/Essentials/blob/993d7ede5664e57a672c6dd7b7062ef31cadafdb/Essentials/src/main/resources/messages.properties#L1505C1-L1509C66

https://github.com/EssentialsX/Essentials/blob/993d7ede5664e57a672c6dd7b7062ef31cadafdb/Essentials/src/main/java/com/earth2me/essentials/commands/Commandessentials.java#L801C1-L810C62

Reply:
Không cần thiết vì:
- Để giảm tải
- Dotman chỉ có server Việt xài
- Mình đưa vào config để admin server có nhu cầu thay đổi các message mà member thấy được. Với các message của ACP thì chỉ cần rõ ràng, đọc hiểu được là được, không cần phải config.

Reply:
❌:
Admin cũng là người dùng, nên cho họ chỉnh sửa để trang trí hoặc tuỳ chỉnh lại cho phù hợp cho từng người.

Reply:
@tranlong5252 

> Admin cũng là người dùng, nên cho họ chỉnh sửa để trang trí hoặc tuỳ chỉnh lại cho phù hợp cho từng người.

Việc đưa admin messages vào config sẽ là sự lãng phí vì:
- DotMan chỉ sử dụng ở Việt Nam => không có nhu cầu dịch sang ngôn ngữ khác.
- Nhu cầu điều chỉnh admin messages là rất hiếm. Các Admin thường chỉ có nhu cầu điều chỉnh cái mà player thấy, thường không cần điều chỉnh cái mà bản thân họ thấy.
  - Ví dụ như Essentials: Ngoại trừ việc áp dụng config có sẵn trên mạng, anh cho rằng các Admin thường chỉ điều chỉnh những message mà hiển thị cho player bình thường thấy.
- Nếu đưa vào config, thì để đảm bảo tính đồng nhất, toàn bộ admin messages khác cũng phải chủ trương đưa vào config => thay đổi workflow, phát sinh công việc không cần thiết.

Reply:
Vậy có gì sẽ tuỳ chỉnh lại các thông báo trước đây sau